contracts/multisig-prover/src/contract.rs

```diff
@@ -250,10 +250,15 @@ mod tests {
         WorkerSet {
             signers: operators
                 .into_iter()
-                .map(|op| Signer {
-                    address: op.address,
-                    pub_key: op.pub_key,
-                    weight: op.weight,
+                .map(|op| {
+                    (
+                        op.address.clone().to_string(),
+                        Signer {
+                            address: op.address,
```

Comment:
Address is stored twice now though. Is that necessary?

Reply:
It's not necessary, but it allows the map to be a `<String, Signer>` instead of `<String, (Uint256, PublicKey)>` which is more readable. I can also remove the `address` from the `Signer` struct, but it feels like it's more understandable to keep all properties of signers into that one struct for future code readability.

What's your thoughts on that?

Reply:
Readability is good, but we don't want to sacrifice performance and store things twice just for the sake of readability. `Signer` is used in a lot of places, so I don't know that removing `address` from `Signer` makes sense. I think just the tuple is okay, but if you want it to be more readable, you can define a new type that is just a (weight, pub key) pair.